src/view/uniprotkb/components/protvista-config/variationFilters.ts

```diff
@@ -0,0 +1,215 @@
+import { scaleLinear } from 'd3-scale';
+import { groupBy } from 'lodash';
+import { TransformedProtvistaVariant } from '../VariationView';
+
+type VariationVariants = { variants: TransformedProtvistaVariant[] }[];
+
+const scaleColors = {
+  UPDiseaseColor: '#990000',
+  UPNonDiseaseColor: '#99cc00',
+  deleteriousColor: '#002594',
+  benignColor: '#8FE3FF',
+  othersColor: '#009e73',
+};
+
+const consequences = {
+  likelyDisease: [/disease/i, /pathogenic\b/i, /risk factor/i],
+  likelyBenign: [/benign/i],
+  uncertain: [/uncertain/i, /conflicting/i, /unclassified/i],
+};
+
+const significanceMatches = (values: RegExp[], clinicalSignificance?: string) =>
+  clinicalSignificance && values.some((rx) => rx.test(clinicalSignificance));
+
+const getFilteredVariants = (
+  variants: VariationVariants,
+  callbackFilter: (variant: TransformedProtvistaVariant) => void
+) =>
+  variants.map((variant) => {
+    const matchingVariants = variant.variants.filter((variantPos) =>
+      callbackFilter(variantPos)
+    );
+    return {
+      ...variant,
```

Comment:
It's a bit convoluted, but a variant has a list of variants (the ones which are collocated)

Reply:
I am thinking that because `variant = { variants: TransformedProtvistaVariant[] }` when we first spread `variant` (which will add its `variants` key & value to the new object) this is just reassigned when we assign `variants: [...matchingVariants]` in the new object.

```	
return {
  ...variant,
  variants: [...matchingVariants],
}
```
is equivalent to:
```	
return {
  variants: [...matchingVariants],
}
```

Reply:
You're right, removed